class/class3.2/src/Color.cpp

```diff
@@ -0,0 +1,22 @@
+#include "Color.hpp"
+
+Color::Color(int r, int g, int b) : r(r), g(g), b(b) {}
+
+Color Color::generateMagicColor() const {
+    int newR = r / 2 - 1;
```

Comment:
there should be some validation that r, g, b components are in range 0..255. For this function you can for example use std::clamp to handle out of range results. but anyway some isValid function is needed